src/lakefs_spec/util.py

```diff
@@ -36,7 +40,7 @@ def md5_checksum(lpath: str, blocksize: int = 2**22) -> str:
     return file_hash.hexdigest()
 
 
-def parse(path: str) -> tuple[str, str, str]:
+def parse(path: Union[str, os.PathLike[str], Path]) -> tuple[str, str, str]:
```

Comment:
This lives behind the API boundary, so its signature should not be changed. Please revert.